src/JBMultiTerminal.sol

```diff
@@ -1370,27 +1411,35 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
     /// @notice Process any fees that are being held for the project.
     /// @param projectId The ID of the project to process held fees for.
     /// @param token The token to process held fees for.
+    /// @param count The number of fees to process.
     /// @param forced If locked held fees should be force processed.
-    function _processHeldFeesOf(uint256 projectId, address token, bool forced) internal {
+    function _processHeldFeesOf(uint256 projectId, address token, uint256 count, bool forced) internal {
+        // Keep a reference to the start index.
+        uint256 startIndex = _nextHeldFeeIndexOf[projectId][token];
+
         // Get a reference to the project's held fees.
         JBFee[] memory heldFees = _heldFeesOf[projectId][token];
```

Comment:
will do

Reply:
Agree! 
The current approach may even result in DoS, because again we don't have a cap on `_heldFeesOf `, which means we will try to make an `SLOAD` for each element in the array (potentially hitting the block gas limit).

The same should be changed also in `_returnHeldFees` and `heldFeesOf`

Reply:
nice! done.